src/eradius_client.erl

```diff
@@ -125,42 +126,90 @@ send_remote_request(Node, {IP, Port, Secret}, Request, Options) when ?GOOD_CMD(R
                         {'DOWN', SenderMonitor, process, SenderPid, _Reason} ->
                             {error, socket_down}
                     end,
-            proceed_response(Request, Response, Peer, TS1, MetricsInfo)
+            proceed_response(Request, Response, Peer, TS1, MetricsInfo, Options)
     catch
         exit:{{nodedown, Node}, _} ->
             {error, node_down}
     end;
 send_remote_request(_Node, {_IP, _Port, _Secret}, _Request, _Options) ->
     error(badarg).
 
-proceed_response(Request, {ok, Response, Secret, Authenticator}, _Peer = {_ServerName, {ServerIP, Port}}, TS1, MetricsInfo) ->
+restore_upstream_server({ServerIP, Port, Retries, InitialRetries}) ->
+    ets:insert(?MODULE, {{ServerIP, Port}, Retries, InitialRetries}).
+
+proceed_response(Request, {ok, Response, Secret, Authenticator}, _Peer = {_ServerName, {ServerIP, Port}}, TS1, MetricsInfo, Options) ->
     update_client_request(Request#radius_request.cmd, MetricsInfo, eradius_lib:timestamp(milli_seconds) - TS1, Request),
     update_client_responses(MetricsInfo),
     case eradius_lib:decode_request(Response, Secret, Authenticator) of
         {bad_pdu, "Message-Authenticator Attribute is invalid" = Reason} ->
             update_client_response(bad_authenticator, MetricsInfo, Request),
             ?LOG(error, "~s INF: Noreply for request ~p. Could not decode the request, reason: ~s", [printable_peer(ServerIP, Port), Request, Reason]),
-            noreply;
+            maybe_failover(Request, noreply, {ServerIP, Port}, Options);
```

Comment:
this is IMHO not a fail-over reason, a misconfigured shared secret is a local problem and should cause an immediate abort of the request.

even if we consider a fail-over for this problem, it should be an immediate action. As I understand the logic, it would only fail over after exhausting the retries counter.

Reply:
fixed